[Sun Ruoxin] iP

vantemoon opened the pull request:
# Jarvis
> "Surprisingly I'm **NOT** related to that guy with a yellow stone on his head 🤡  "

Jarvis is here to help you remember every task you need to take note of.
- text-based and super easy to use ‼️ 
- different types of tasks supported

Getting started now 💥 
1. Download it from [here](https://github.com/vantemoon/ip)
2. Double-click it
3. And enjoy stress-free task management 🤩 

*Hello?* What are you waiting for?

Features:
- [x] Managing to-dos
- [x] Managing deadlines
- [x] Managing events
- [ ] Reminders (coming soon)

---
If you are also a Java programmer, feel free to use it to practice Java too. 
Here's the `main` method:
``` java
public static void main(String[] args) throws DukeException {
        new Duke("data/tasks.txt").run();
    }
```

Commit:
Add Gradle support

vantemoon committed:
Renamed the chatbot and modified the greeting message

vantemoon committed:
Renamed the chatbot and modified the greeting message

vantemoon committed:
Implemented echo and exit with modified output format

vantemoon committed:
Implemented echo and exit with modified output format

vantemoon committed:
Added the ability to store user input

vantemoon committed:
Added the ability to store user input

vantemoon committed:
Added the ability to display the list of inputs back to the user when requested

vantemoon committed:
Added the ability to display the list of inputs back to the user when requested

vantemoon committed:
Added a Task class to represent tasks

vantemoon committed:
Added a Task class to represent tasks

vantemoon committed:
Added the ability to mark tasks as done or not done.

vantemoon committed:
Added the ability to mark tasks as done or not done.

vantemoon committed:
Added toString method for task

vantemoon committed:
Added toString method for task

vantemoon committed:
Added separate classes for todos, deadlines, and events

vantemoon committed:
Added separate classes for todos, deadlines, and events

vantemoon committed:
Implemented semi-automatic testing using the I/O redirection technique

vantemoon committed:
Implemented semi-automatic testing using the I/O redirection technique

vantemoon committed:
Created exception class to handle error

vantemoon committed:
Created exception class to handle error

vantemoon committed:
Added exception handling

vantemoon committed:
Added exception handling

vantemoon committed:
Added support for deleting tasks from the list.

vantemoon committed:
Added support for deleting tasks from the list.

vantemoon committed:
Change the name of print method parameters

vantemoon committed:
Change the name of print method parameters

vantemoon committed:
Add the ability to save tasks in the hard disk automatically whenever the task list changes

vantemoon committed:
Add the ability to save tasks in the hard disk automatically whenever the task list changes

vantemoon committed:
Add the ability to store deadline dates as a java.time.LocalDate

vantemoon committed:
Add the ability to store deadline dates as a java.time.LocalDate

vantemoon committed:
Merge branch 'branch-Level-8'

vantemoon committed:
Extract out closely related code as classes

vantemoon committed:
Extract out closely related code as classes

vantemoon committed:
Refactor the code to extracted classes

vantemoon committed:
Refactor the code to extracted classes

vantemoon committed:
Organise the classes into suitable java packages

vantemoon committed:
Organise the classes into suitable java packages

vantemoon committed:
Merge commit 'b3a93868925ffb1f3d6316d58a61ae1a6650ee59'

vantemoon committed:
Use Gradle to automate some of the build tasks of the project

vantemoon committed:
Add JUnit tests to test the behaviour of the code

vantemoon committed:
Edit JUnit test method name

vantemoon committed:
Add Javadoc comments to Duke.java and Ui.java

vantemoon committed:
Add Javadoc comments to more classes

vantemoon committed:
Add header comments to all non-private classes/methods

vantemoon committed:
Tweak the code to comply with the given coding standard

vantemoon committed:
Add the ability to find a task by searching for a keyword

vantemoon committed:
Merge branch 'branch-A-CodingStandard'

vantemoon committed:
Merge branch 'branch-Level-9'

vantemoon committed:
Build basic GUI structure for Jarvis

vantemoon committed:
Rebuild the functionalities to work with the GUI

AugustDespair commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,47 @@
+     * @return "T" representing to-do
+     */
+    @Override
+    public String getType() {
```
I think this could be a static variable instead, since all Todo classes share the same Type, which is 'T'.

vantemoon replied:
Hi, thanks for your suggestion! I do get your point on how all Todo instances return the same 'T' thus the method should be static. However, since the getType() methods in Todo, Deadline and Event are overwriting the abstract method in Task, they cannot be static as overriding depends on dynamic binding:(

AugustDespair commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,58 @@
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) {
+        ArrayList<Task> matches = tasks.find(keyword);
+//        ui.findMessage(matches);
```
If I remember correctly, all comments should be indented based on the line of code below it, so the double slash should be right next to `ui.findMessage(matches);`

AugustDespair commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,54 @@
+ */
+public class Event extends Task {
+    /** The time of the event. */
+    protected String at;
```
I think a variable name like `eventTime` would be more descriptive than `at`.

o-ohst replied:
I agree that `eventTime` would be a more appropriate name.

AugustDespair commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,73 @@
+ */
+public class Deadline extends Task {
+    /** The date of the deadline. */
+    protected LocalDate by;
```
I think having the variable name be `deadline` or `deadlineDate` would be more descriptive than `by`.

o-ohst replied:
I agree that `deadline` would be a more appropriate name.

AugustDespair reviewed (commented):
Your code is really organized and clean. Honestly, most of my comments are just nitpicks.

vantemoon committed:
Add main class for GUI

vantemoon committed:
Rebuild the Scene using FXML and remove text-based UI

vantemoon committed:
Add Javadoc comments

Some of the newly added classes/methods were missing Javadoc.
Adding header comments to those improves the coding style.

vantemoon committed:
Add assertions to verify important assumptions

vantemoon committed:
Remove unnecessary comments

Some comments in the code are unnecessary as the code itself is
self-explanatory.

Removing these comments improves the code quality.

Let's delete unnecessary comments from the code.

vantemoon committed:
Replace magic number in checkStatus() method

In the checkStatus() method, integer 1 means the task is done and 0
means the task is not done yet. Simply compare the parameter with an
integer can be confusing as others don't know what's the meaning
behind the integer.

Using a named constant to represent the status of a completed task
and compare the parameter to the constant is easier to understand
because the name tells us more about the meaning of the number.

Let's declare a new constant DONE to represent the status of a completed
task and compare the parameter to DONE to determine whether the task is
completed or not.

vantemoon committed:
Merge pull request #1 from vantemoon/branch-A-Assertions

Add assertions to verify important assumptions

vantemoon committed:
Merge branch 'master' into branch-A-CodeQuality

vantemoon committed:
Merge pull request #2 from vantemoon/branch-A-CodeQuality

Amend the code to improve code quality

vantemoon committed:
Use GitHub Actions to set up CI

vantemoon committed:
Add the ability to clone tasks

vantemoon committed:
Fix coding style errors

vantemoon committed:
Improve code quality

The parse() method in class Parser is too long thus has poor readability.
Extract out some code blocks can make it easier to be understand.

To improve the readability of the code, each case (different types of
command) is extracted out into a separate method.

vantemoon committed:
Fix CheckStyle errors

vantemoon committed:
Fix bugs when load from local file

vantemoon committed:
Add a screenshot of the product

vantemoon committed:
Fix gradle config

vantemoon committed:
Fix checkStyle errors

vantemoon committed:
Update README.md

vantemoon committed:
Fix image source path
